src/Client.php

```diff
@@ -68,4 +68,11 @@ public function getTypes()
     {
         return $this->encoder->__getTypes();
     }
+
+    public function overrideLocation($location)
+    {
+        $copy = clone $this;
+        $copy->encoder = $this->encoder->overrideLocation($location);
+        return $copy;
+    }
```

Comment:
Done and done. I find that withLocation can be confusing as we're manipulating different locations here. I chose withOverridenTarget(), but withTarget is fine by me too. Let me know what you prefer.